drivers/soundwire/intel.c

```diff
@@ -1300,66 +1300,66 @@ static int intel_init(struct sdw_intel *sdw)
 /*
  * probe and init
  */
-static int intel_master_probe(struct sdw_master_device *md, void *link_ctx)
+static int intel_master_probe(struct platform_device *pdev)
 {
+	struct sdw_intel_link_res *parent_res;
 	struct sdw_intel *sdw;
 	int ret;
 
-	sdw = devm_kzalloc(&md->dev, sizeof(*sdw), GFP_KERNEL);
+	sdw = devm_kzalloc(&pdev->dev, sizeof(*sdw), GFP_KERNEL);
 	if (!sdw)
 		return -ENOMEM;
 
-	sdw->instance = md->link_id;
-	sdw->link_res = link_ctx;
-	sdw->cdns.dev = &md->dev;
+	sdw->instance = pdev->id;
+	sdw->link_res = dev_get_platdata(&pdev->dev);
+	sdw->cdns.dev = &pdev->dev;
 	sdw->cdns.registers = sdw->link_res->registers;
-	sdw->cdns.instance = md->link_id;
+	sdw->cdns.instance = pdev->id;
 	sdw->cdns.msg_count = 0;
-	sdw->cdns.bus.dev = &md->dev;
-	sdw->cdns.bus.link_id = md->link_id;
-	sdw->link_res->cdns = &sdw->cdns;
+	sdw->cdns.bus.dev = &pdev->dev;
+	sdw->cdns.bus.link_id = pdev->id;
 
 	sdw_cdns_probe(&sdw->cdns);
 
 	/* Set property read ops */
 	sdw_intel_ops.read_prop = intel_prop_read;
 	sdw->cdns.bus.ops = &sdw_intel_ops;
 
-	md->pdata = sdw;
-
 	/* set driver data, accessed by snd_soc_dai_set_drvdata() */
-	dev_set_drvdata(&md->dev, &sdw->cdns);
+	dev_set_drvdata(&pdev->dev, &sdw->cdns);
 
 	/* use generic bandwidth allocation algorithm */
 	sdw->cdns.bus.compute_params = sdw_compute_params;
 
 	ret = sdw_add_bus_master(&sdw->cdns.bus);
 	if (ret) {
-		dev_err(&md->dev, "sdw_add_bus_master fail: %d\n", ret);
+		dev_err(&pdev->dev, "sdw_add_bus_master fail: %d\n", ret);
 		return ret;
 	}
 
-	if (sdw->cdns.bus.prop.hw_disabled)
-		dev_info(&md->dev,
-			 "SoundWire master %d is disabled, will be ignored\n",
+	if (sdw->cdns.bus.prop.hw_disabled) {
+		dev_info(&pdev->dev, "SoundWire master %d is disabled, ignoring\n",
 			 sdw->cdns.bus.link_id);
+	}
```

Comment:
@plbossart the parentheses are superfluous?

Reply:
yes, I could drop them. Let's see what the test results are first before we look into style...

Reply:
will do this in follow-up cleanup